diff --git a/.gitignore b/.gitignore
index e733d35..43c24f9 100644
--- ./.gitignore
+++ ./.gitignore
@@ -45,3 +45,7 @@ mkinstalldirs
 ylwrap
 configure
 aclocal.m4
+src/cscope
+src/egrep.c
+src/egrep.h
+src/fscanner.c
diff --git a/build_prerequisite.sh b/build_prerequisite.sh
new file mode 100755
index 0000000..7780973
--- /dev/null
+++ ./build_prerequisite.sh
@@ -0,0 +1,26 @@
+#!/bin/zsh
+SCRIPT=$(realpath "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+cd "$SCRIPTPATH"
+
+if [ $(uname) != "Darwin" ]
+then
+    apt-get update
+    apt-get install -y \
+        automake \
+        autotools-dev \
+        bison \
+        byacc \
+        flex \
+        libncurses5-dev \
+        libncursesw5-dev
+
+else
+    brew install \
+        autoconf \
+        automake \
+        bison \
+        byacc \
+        flex \
+        libtool
+fi
diff --git a/do_build.sh b/do_build.sh
new file mode 100755
index 0000000..bcf5fe1
--- /dev/null
+++ ./do_build.sh
@@ -0,0 +1,13 @@
+#!/bin/zsh
+SCRIPT=$(realpath "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+cd "$SCRIPTPATH"
+
+./build_prerequisite.sh
+
+aclocal && \
+    autoconf && \
+    automake && \
+    ./configure && \
+    make && \
+    make install
diff --git a/src/global.h b/src/global.h
index a6f1486..7be1874 100644
--- ./src/global.h
+++ ./src/global.h
@@ -416,4 +416,5 @@ int	execute(char *a, ...);
 long	dbseek(long offset);
 
 
+#define BUFSIZ  1024        /* size of buffer used by setbuf */
 #endif /* CSCOPE_GLOBAL_H */
