diff --git .gitignore .gitignore
index e733d35..0f91e47 100644
--- .gitignore
+++ .gitignore
@@ -45,3 +45,14 @@ mkinstalldirs
 ylwrap
 configure
 aclocal.m4
+src/cscope
+Makefile.in
+contrib/Makefile.in
+doc/Makefile.in
+src/Makefile.in
+src/egrep.c
+src/egrep.h
+src/fscanner.c
+config.h.in
+configure
+.env.json
diff --git src/global.h src/global.h
index a6f1486..7be1874 100644
--- src/global.h
+++ src/global.h
@@ -416,4 +416,5 @@ int	execute(char *a, ...);
 long	dbseek(long offset);
 
 
+#define BUFSIZ  1024        /* size of buffer used by setbuf */
 #endif /* CSCOPE_GLOBAL_H */
